src/protocols/hpr_protocol_router.erl

```diff
@@ -192,6 +132,117 @@ terminate(_Reason, _State = #state{}) ->
 %% Internal Function Definitions
 %% ------------------------------------------------------------------
 
+-spec get_stream(
+    Gateway :: libp2p_crypto:pubkey_bin(),
+    LNS :: binary(),
+    Server :: hpr_route:server(),
+    Backoff :: backoff:backoff()
+) -> {ok, grpcbox_client:stream()} | {error, any()}.
+get_stream(Gateway, LNS, Server, Backoff0) ->
+    {Delay, BackoffFailed} = backoff:fail(Backoff0),
+    case ets:lookup(?STREAM_ETS, ?KEY(Gateway, LNS)) of
+        [{_, ?CONNECTING}] ->
+            timer:sleep(Delay),
+            get_stream(Gateway, LNS, Server, BackoffFailed);
+        [{_, #{channel := ChannelPid, stream_pid := StreamPid} = Stream}] ->
+            case erlang:is_process_alive(ChannelPid) andalso erlang:is_process_alive(StreamPid) of
+                true ->
+                    {ok, Stream};
+                false ->
+                    true = ets:delete(?STREAM_ETS, ?KEY(Gateway, LNS)),
+                    get_stream(Gateway, LNS, Server, BackoffFailed)
+            end;
+        [] ->
+            true = ets:insert(?STREAM_ETS, {?KEY(Gateway, LNS), ?CONNECTING}),
```

Comment:
Use insert_new here and try again if it fails:

https://github.com/helium/blockchain-core/blob/master/src/ledger/v1/blockchain_ledger_v1.erl#L743

Reply:
Thx, this should improve in case I get 2 at the exact same time, This is why I added some random jitter in test...